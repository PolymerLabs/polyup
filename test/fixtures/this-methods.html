<link rel="import" href="../polymer/polymer.html">

<polymer-element name="this-methods-element">
  <script>
    Polymer({
      ready: function() {
        this.job('doThing', this.doThing, 300);
      },
      doThing: function() {
        var imagePath = this.resolvePath('image.png');
        console.log(imagePath);
        this.imagePath = imagePath;
        this.deep.prop = 2 * 2;
        this.deep[prop].of.another.place[toGo + '1'] = 2 * 2;
        // Don't upgrade expressions inside inner functions
        function innerStatement () {
          this.imagePath = imagePath;
        }
        var innerExpression = function() {
          this.imagePath = imagePath;
        };
      }
    });
  </script>
</polymer-element>
