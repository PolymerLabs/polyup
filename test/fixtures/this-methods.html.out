<link rel="import" href="../polymer/polymer.html">

<dom-module id="this-methods-element">
  <script>
    Polymer({
      is: 'this-methods-element',
      ready: function () {
        this.debounce('doThing', this.doThing, 300);
      },
      doThing: function () {
        var imagePath = this.resolveUrl('image.png');
        console.log(imagePath);
      }
    });
  </script>
</dom-module>
